### ISSUE TITLE
Show auto-expand checkboxes as disabled when prefetch is disabled

### ISSUE THREAD

**Issue by YaManicKill** [thelounge/lounge#1226]
If prefetch is disabled on the server, then we completely hide the auto-expand checkboxes. In at least 1 instance, someone has been confused about why they can't enable this, and I wasn't even sure why it wasn't there. I had to check, and it turns out their prefetch wasn't enabled.

I think we should show the checkboxes, but make them disabled (and off) and show a message explaining that their server admin has turned this feature off, and they need to get them to enable it if they want it.

I think that's much better than not showing the option at all.

**Comment by PolarizedIons** [comment]
Is this still an issue? Seems like now the options are only shown if prefetch is enabled

https://github.com/thelounge/lounge/blob/fc0af518c6dd5251123d71f54d702453ab08f6eb/client/index.html#L272-L288

**Comment by YaManicKill** [comment]
Nope, sorry. I misread the issue. We currently hide them. The issue is to not hide them, but show them as disabled, with an explanation.

**Comment by PolarizedIons** [comment]
Ah, also misread that a bit then

**Comment by astorije** [comment]
I have to say I'm torn on that one 😅

In general, good UX pattern when it comes to disabled vs. hidden feature is: can the user do anything about it? If answer is yes, give them incentive that this is around the corner with an actionable message, like a "Turn this on in the settings" tooltip or such. If answer is no, then there is no reason to frustrate a user by telling them "this is what you _could_ do except you can't and there is nothing you can do about it" :D

In this case, it's... both: no indeed, there is nothing the user can do _themselves_. But yes indeed, they _can_ contact their admin to turn it on. Except as an admin, do I want to get pinged by everyone in my team to enable this? :D
And if admin says no, we're back to square one, there is nothing the user can do.

It seems to me that what this is about is if someone wants to use that feature but forgot to enable it and didn't realize that. In this case, a warning/info log at server startup would seem more appropriate to me, and keep the current state on the client as it is.

Happy to discuss, as said I'm not 100% convinced on either solution, so we can definitely discuss options if what I just said is not totally clear or reasonable.

**Comment by YaManicKill** [comment]
Well, the reason this was created was because @TheSonAlsoRises actually got confused about this because he didn't understand why it wasn't there.

Some of the users *can* do something about it, and some can't do anything directly. But they can do *something* about it, even if that doesn't necessarily mean that it will end up enabled.

> In this case, a warning/info log at server startup would seem more appropriate to me, and keep the current state on the client as it is.

What? We have a warning/info log every time someone starts up a server that doesn't have it enabled? The setting is disabled by default. This is definitely wrong.

There are plenty of examples of where this is a thing done. If we take slack's irc gateway for example. You can always go into the settings and see the setting exists, but a warning that only admins can turn it on. So you can go ask your admin to enable it. Otherwise, how do you ever know (as a user) that this is even a possible feature?

**Comment by astorije** [comment]
> Well, the reason this was created was because @TheSonAlsoRises actually got confused about this because he didn't understand why it wasn't there.

Not trying to undermine anyone here, but if a single user reported this / got confused by this, then maybe a more appropriate (or additional) response would be better documentation. I'll make sure to make this clearer as I get back into rewriting the docs.

> This is definitely wrong.

How so? It's a sensible debug/info message to display at server startup (`"[INFO] Link previews are disabled on this instance` or such), _especially_ if it's disabled by default!

> Otherwise, how do you ever know (as a user) that this is even a possible feature?

That's the thing though, one could argue that if an admin disabled it (or didn't enable it, in our case), and did it on purpose, then technically it is not a possible feature.

---

But as I said, I'm on the fence and would be okay if someone came with a PR that does your original suggestion. Just trying to weigh both sides of the argument before we address it :)

**Comment by YaManicKill** [comment]
> How so? It's a sensible debug/info message to display at server startup ("[INFO] Link previews are disabled on this instance or such), especially if it's disabled by default!

No, because we don't want to do that for every disabled feature. That's silly.

> That's the thing though, one could argue that if an admin disabled it (or didn't enable it, in our case), and did it on purpose, then technically it is not a possible feature.

... 

Of course it's a possible feature. It's just disabled. You are misusing language here. Sure, a user that isn't an admin can't enable it, it's not possible for *them* to enable it, but it is still possible to be done.

> Not trying to undermine anyone here, but if a single user reported this / got confused by this, then maybe a more appropriate (or additional) response would be better documentation. 

Documentation isn't a replacement for good UX. We disagree on what the correct UX is in this situation. But if someone has to go to the documentation to know this feature exists (and that they have to enable it on the server) then I think that is bad. 

I mean, we shouldn't be requiring admins to go edit a text file to enable it *anyway*, but that's a different discussion.
